ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/DefaultP2PNetwork.java

```diff
@@ -265,6 +265,7 @@ public void awaitStop() {
   @Override
   public boolean addMaintainConnectionPeer(final Peer peer) {
     final boolean wasAdded = maintainedPeers.add(peer);
+    peerDiscoveryAgent.bond(peer);
```

Comment:
Does this mean that we require discovery to be enabled on peers that we want to add using static-nodes.json file or AddPeer API? Because I feel like most uses that are statically mapping their peers might not have discovery enabled (and might not want to).

Reply:
No - we try to create a `DiscoveryPeer` [to initiate bonding with](https://github.com/mbaxter/besu/blob/35436d777a1eec5bb3d8522a350c5c5a3f06a376/ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgent.java#L311-L317).  The [`DiscoveryPeer.from` helper](https://github.com/mbaxter/besu/blob/35436d777a1eec5bb3d8522a350c5c5a3f06a376/ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/DiscoveryPeer.java#L58) will only return a peer if discovery appears to be enabled.  The bonding process consists of sending a `PING` to that peer, and waiting for a response.  Only if the peer is running discovery and responds do we add the peer to our table.  If the peer doesn't respond, we'll retry a few times and then stop.

Users can explicitly indicate that discovery is disabled by supplying enode urls with [`discport=0`](https://github.com/mbaxter/besu/blob/35436d777a1eec5bb3d8522a350c5c5a3f06a376/ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/peers/EnodeURLTest.java#L152-L167), which will cause us to skip the bonding attempt.